### ISSUE TITLE
Typescript compilation errors in Knockouts index.d.ts

### ISSUE THREAD

**Issue by dhilgarth** [DevExpress/testcafe#1755]
### Are you requesting a feature or reporting a bug?

Bug

### What is the current behavior?

I have added testcafe to my Typescript / KnockoutJS web app via `npm install --save-dev testcafe`.
I also added `"test": "testcafe chrome tests/"` to `"scripts"` in package.json.

I have added a folder called `tests` and added a file called `my-test.ts`.

Now, when I execute `npm test` I am getting a lot of TypeScript compilation errors from `node_modules/@types/knockout/index.d.ts`.

### What is the expected behavior?

No compilation errors

### How would you reproduce the current behavior (if this is a bug)?

Simply add testcafe to a Knockout / Typescript application

### Specify your

* operating system: Windows 10
* testcafe version: 0.17.1
* node.js version: 6.10.2

**Comment by AlexanderMoskovkin** [comment]
Hi @dhilgarth,

I tried to reproduce the issue on a simple example but unsuccessfully. I've cloned the [Microsoft/TypeScript-Knockout-Starter](https://github.com/AlexanderMoskovkin/TypeScript-Knockout-Starter/tree/gh-1755) and have added a file with TestCafe test there. It works properly on my side.
Could you please modify my example or create yours to reproduce the problem?

**Comment by dhilgarth** [comment]
You should be able to reproduce it with these commands:

```
npm uninstall -S @types/knockout
npm install -D @types/knockout@3.4.40
```

**Comment by AlexanderMoskovkin** [comment]
Thanks for the clarification. I've reproduced the issue. We've already fixed it in the context of the #1537 issue. It will be available in the next version (we're going to release it in a few days).
As a workaround you can modify the file: `../node_modules/testcafe/lib/compiler/test-file/formats/typescript.js`. Add the following fix: https://github.com/DevExpress/testcafe/commit/d1ee1016fa123893dc9c44856112f6468b32561d

Also the fix is already available in the latest `dev` version: `testcafe@0.17.2-dev20170831`. But it has a couple of know issues: there are problems with browser actions (like resize, takeScreenshot) on MacOS and Linux. We're working on fixes.

**Comment by dhilgarth** [comment]
Great, thanks for the followup. My workaround for now is to update to the latest version of `@types/knockout`. This fixed it for me as well.
